packages/components/src/DateTimePickers/InputDateTimePicker/InputDateTimePicker.component.js

```diff
@@ -0,0 +1,221 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import getMinutes from 'date-fns/get_minutes';
+import getHours from 'date-fns/get_hours';
+import getDate from 'date-fns/get_date';
+import getMonth from 'date-fns/get_month';
+import getYear from 'date-fns/get_year';
+import setDate from 'date-fns/set_date';
+import setMinutes from 'date-fns/set_minutes';
+import lastDayOfMonth from 'date-fns/last_day_of_month';
+import { twoDigits } from '../pickers/TimePicker';
+import DateTimePicker from '../DateTimePicker';
+import theme from './InputDateTimePicker.scss';
+
+const splitDateAndTimeRegex = new RegExp(/^(.*),(.*)$/);
+const dateRegex = new RegExp(/^\s*([0-9]{1,2})\/([0-9]{1,2})\/([0-9]{4})\s*$/);
+const timeRegex = new RegExp(/^\s*([0-9]{1,2}):([0-9]{2})\s*$/);
+
+function hoursAndMinutesToTime(hours, minutes) {
+	return hours * 60 + minutes;
+}
+
+function getTextDate(date, time) {
+	if (date === undefined) {
+		return '';
+	}
+
+	const day = getDate(date);
+	const month = getMonth(date) + 1;
+	const year = getYear(date);
+
+	const dateText = `${twoDigits(day)}/${twoDigits(month)}/${year}`;
+
+	if (time === undefined) {
+		return dateText;
+	}
+
+	const hours = Math.floor(time / 60);
+	const minutes = time % 60;
+
+	const timeText = `, ${twoDigits(hours)}:${twoDigits(minutes)}`;
+
+	return dateText + timeText;
+}
+
+
+class InputDateTimePicker extends React.Component {
+	static propTypes = {
+		selectedDateTime: PropTypes.instanceOf(Date),
+		onChange: PropTypes.func.isRequired,
```

Comment:
This is an input, so it should follow the same propTypes.
* `value`: string
* `onChange`: func no required

Reply:
I was not thinking of it as an input but more like a thing that take a selected date and give a date.
The date is the final value wanted, not really the string.
We need to discuss about it, it depends on how we reason and manage date in our frontend programs and eventually how we convert it from/to the server API.
But a date object is more useful to do manipulations and I don't think server API should dictate the frontend for the format (we'll just eventually need a date parser/stringifier).

Reply:
✘